python/tvm/driver/tvmc/runner.py

```diff
@@ -337,135 +347,75 @@ def run_module(
-
-        return outputs, times
-
-
-def get_top_results(outputs, max_results):
-    """Return the top n results from the output tensor.
-
-    This function is primarily for image classification and will
-    not necessarily generalise.
-
-    Parameters
-    ----------
-    outputs : dict
-        Outputs dictionary - {output_name: np.array}.
-    max_results : int
-        Number of results to return
-
-    Returns
-    -------
-    top_results : np.array
-        Results array of shape (2, n).
-        The first row is the indices and the second is the values.
-
-    """
-    output = np.copy(outputs["output_0"])
-    sorted_labels = output.argsort()[0][-max_results:][::-1]
-    output.sort()
-    sorted_values = output[0][-max_results:][::-1]
-    top_results = np.array([sorted_labels, sorted_values])
-    return top_results
-
-
-def format_times(times):
-    """Format the mean, max, min and std of the execution times.
-
-    This has the effect of producing a small table that looks like:
-
-        Execution time summary:
-        mean (ms)   max (ms)    min (ms)    std (ms)
-        0.14310    0.16161    0.12933    0.01004
-
-    Parameters
-    ----------
-    times : list
-        A list of execution times (in seconds).
-
-    Returns
-    -------
-    str
-        A formatted string containing the statistics.
-    """
-
-    # timestamps
-    mean_ts = np.mean(times) * 1000
-    std_ts = np.std(times) * 1000
-    max_ts = np.max(times) * 1000
-    min_ts = np.min(times) * 1000
-
-    header = "Execution time summary:\n{0:^10} {1:^10} {2:^10} {3:^10}".format(
-        "mean (ms)", "max (ms)", "min (ms)", "std (ms)"
-    )
-    stats = "{0:^10.2f} {1:^10.2f} {2:^10.2f} {3:^10.2f}".format(mean_ts, max_ts, min_ts, std_ts)
+            logger.debug("running on remote RPC with no key")
+            session = rpc.connect(hostname, port)
+    else:
+        # Local
+        logger.debug("running a local session")
+        session = rpc.LocalSession()
+
+    session.upload(lib_path)
+    lib = session.load_module(f"{tvmc_package.name}.so")
+
+    # TODO expand to other supported devices, as listed in tvm.rpc.client (@leandron)
+    logger.debug("device is %s", device)
+    if device == "gpu":
+        dev = session.gpu()
+    elif device == "cl":
+        dev = session.cl()
+    else:
+        assert device == "cpu"
+        dev = session.cpu()
 
-    return "%s\n%s\n" % (header, stats)
+    if profile:
+        logger.debug("creating runtime with profiling enabled")
+        module = debug_executor.create(graph, lib, dev, dump_root="./prof")
+    else:
+        logger.debug("creating runtime with profiling disabled")
+        module = runtime.create(graph, lib, dev)
+
+    logger.debug("load params into the runtime module")
+    module.load_params(params)
+
+    shape_dict, dtype_dict = get_input_info(graph, params)
+    inputs_dict = make_inputs_dict(shape_dict, dtype_dict, inputs, fill_mode)
```

Comment:
is it possible to ever recover these inputs?

Reply:
when you run you have the option of providing numpy inputs. If you don't provide it, theyll be randomly generated and arent recoverable. Presumably if you're not giving it inputs you dont care about the outputs and are just doing performance benchmarking.

Reply:
ok, maybe worth including this as a comment here?